pythorhead/community.py

```diff
@@ -86,26 +86,15 @@ def list(  # noqa: A003
         Args:
             limit (Optional[int], optional): Defaults to None.
             page (Optional[int], optional): Defaults to None.
-            sort (Optional[SortType], optional): Defaults to None.
-            type_ (Optional[ListingType], optional): Defaults to None.
+            sort (Optional[str], optional): Defaults to None. (Options: "Hot", "New", "Old", "Active", "TopAll", "TopDay", "TopWeek", "TopMonth", "TopYear", "TopHour", "TopSixHour", "TopTwelveHour", "NewComments", "MostComments")
+            type_ (Optional[str], optional): Defaults to None. (Options: "Local", "Subscribed", "All", "Community")
 
         Returns:
             List[dict]: list of communities
         """
-        list_community: dict = {}
-
-        if limit is not None:
-            list_community["limit"] = limit
-        if page is not None:
-            list_community["page"] = page
-        if sort is not None:
-            list_community["sort"] = sort.value
-        if type_ is not None:
-            list_community["type"] = type_.value
-
-        if data := self._requestor.api(Request.GET, "/community/list", params=list_community):
-            return data["communities"]
-        return []
+   
+        params: list[str, Any] = {key: value for key, value in locals().items() if value is not None and key != "self"}
+        return self._requestor.api(Request.GET, "/community/list", params=params)
```

Comment:
wouldn't this return a dict with a `communities` key, instead of a list of communities?

Reply:
Not entirely sure what you mean sorry, but I believe it returns the same as the original code did. Which is X communities sorted by Y in a dict.

https://join-lemmy.org/api/classes/LemmyHttp.html#listCommunities

Reply:
before your PR:
```python
return data["communities"]
```

If I'm understanding correctly, what your PR does is equivalent to this:
```python
return data
```
this adds an extra level of indirection to the result of this request. Users that used the method previously, will now have to patch their code to get the `data["communities"]` key content for the same result

Reply:
Right I get you now, thank you.

So this would be a simple fix with:
```
        if data := self._requestor.api(Request.GET, "/community/list", params=params):
            return data["communities"]
        return []
```

Right? It works locally for me doing this (I had the communities key already hence why it worked) and updating the code to the above and removing the key seems to pull through the data properly after updating my code to reflect this.